crates/flux-fhir-analysis/src/wf/sortck.rs

```diff
@@ -283,17 +283,27 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
         }
     }
 
-    fn synth_var(&mut self, path: &fhir::PathExpr) -> rty::Sort {
+    fn synth_var(&mut self, path: &fhir::PathExpr) -> Result<rty::Sort> {
         match path.res {
-            ExprRes::Param(_, id) => self.param_sort(id),
+            ExprRes::Param(_, id) => Ok(self.param_sort(id)),
             ExprRes::Const(def_id) => {
-                // TODO(nilehmann) generalize const sorts
-                let ty = self.genv.tcx().type_of(def_id).no_bound_vars().unwrap();
-                assert!(ty.is_integral());
-                rty::Sort::Int
+                if let Some(sort) = self
+                    .genv
+                    .sort_of_def_id(def_id)
+                    .unwrap_or_else(|_err| span_bug!(path.span, "unknown sort for `{def_id:?}`"))
```

Comment:
This error should be emitted

```suggestion
                    .emit(&self.genv)?
```